diff --git a/src/arch.rs b/src/arch.rs
index a998697..384120a 100644
--- a/src/arch.rs
+++ b/src/arch.rs
@@ -30,6 +30,7 @@ pub const EM_SELF: u16 = {
         target_arch = "x86_64" => EM_X86_64,
         target_arch = "x86" => EM_386,
         target_arch = "aarch64" => EM_AARCH64,
+        target_arch = "riscv64" => EM_RISCV,
     }
 };
 
@@ -39,6 +40,7 @@ pub const R_RELATIVE: u32 = {
         target_arch = "x86_64" => R_X86_64_RELATIVE,
         target_arch = "x86" => R_386_RELATIVE,
         target_arch = "aarch64" => R_AARCH64_RELATIVE,
+        target_arch = "riscv64" => R_RISCV_RELATIVE,
     }
 };
 
@@ -48,6 +50,7 @@ pub const NIX_SYSTEM: &str = match option_env!("NIX_SYSTEM") {
         target_arch = "x86_64" => "x86_64_linux",
         target_arch = "x86" => "i686_linux",
         target_arch = "aarch64" => "aarch64_linux",
+        target_arch = "riscv64" => "riscv64_linux",
     },
 };
 
@@ -72,6 +75,9 @@ macro_rules! main_relocate_stack {
             target_arch = "aarch64" => {
                 core::arch::asm!("mov sp, {}; bl {}", in(reg) $sp, sym $func, options(noreturn));
             }
+            target_arch = "riscv64" => {
+                core::arch::asm!("mv sp, {}; call {}", in(reg) $sp, sym $func, options(noreturn));
+            }
         }
     };
 }
@@ -89,6 +95,9 @@ macro_rules! elf_jmp {
             target_arch = "aarch64" => {
                 core::arch::asm!("mov sp, {}; br {}", in(reg) $sp, in(reg) $target, options(noreturn));
             }
+            target_arch = "riscv64" => {
+                core::arch::asm!("mv sp, {}; jr {}", in(reg) $sp, in(reg) $target, options(noreturn));
+            }
         }
     };
 }
diff --git a/src/sys.rs b/src/sys.rs
index 109d809..aca9608 100644
--- a/src/sys.rs
+++ b/src/sys.rs
@@ -182,5 +182,4 @@ pub fn new_slice_leak(size: usize) -> Option<&'static mut [u8]> {
 #[cfg(not(test))]
 #[lang = "eh_personality"]
 #[no_mangle]
-pub extern fn rust_eh_personality() {
-}
+pub extern "C" fn rust_eh_personality() {}
