diff --git a/runtime/loader-binary/loader.S b/runtime/loader-binary/loader.S
index 5908ffa..c69b7e4 100644
--- a/runtime/loader-binary/loader.S
+++ b/runtime/loader-binary/loader.S
@@ -3,6 +3,11 @@
 #include <asm/csr.h>
 #include "util/asm_helpers.h"
 
+.option push
+.option norvc
+.option nopic
+.option norelax
+
 .section .text
 _start:
   // a1: dram_base
@@ -67,3 +72,5 @@ _start:
 
 exit:
   call error_and_exit
+
+.option pop
